src/read/names.rs

```diff
@@ -0,0 +1,2641 @@
+use std::{string::String, vec::Vec};
+
+#[cfg(not(feature = "std"))]
+use alloc::{string::String, vec::Vec};
+
+/// A single parsed name entry from the `.debug_names` section.
+#[derive(Debug, Clone)]
+pub struct NameEntry<R: Reader> {
+    unit_header_offset: DebugInfoOffset<R::Offset>,
+    die_offset: UnitOffset<R::Offset>,
+    name: R,
+}
+
+impl<R: Reader> NameEntry<R> {
+    /// Returns the name this entry refers to.
+    pub fn name(&self) -> &R {
+        &self.name
+    }
+
+    /// Returns the offset into the .debug_info section for the header of the compilation unit
+    /// which contains this name.
+    pub fn unit_header_offset(&self) -> DebugInfoOffset<R::Offset> {
+        self.unit_header_offset
+    }
+
+    /// Returns the offset into the compilation unit for the debugging information entry which
+    /// has this name.
+    pub fn die_offset(&self) -> UnitOffset<R::Offset> {
+        self.die_offset
+    }
+}
+
+/// A parsed entry from the entry pool in a `.debug_names` unit.
+///
+/// This structure represents a fully parsed debug information entry (DIE) reference
+/// from the DWARF 5 debug_names section. Each entry corresponds to a named symbol
+/// in the debugging information and provides efficient access to its location and
+/// relationships within the DWARF data.
+///
+/// According to DWARF 5 Section 6.1.1.4.6, each entry in the entry pool contains:
+/// - An abbreviation code referencing the abbreviation table
+/// - Attributes as described by the abbreviation declaration
+/// - At minimum, a die_offset attribute pointing to the actual DIE
+///
+/// # Example Usage
+/// ```ignore
+/// let parsed_entry = unit.parse_entry_pool_entry(offset, &abbrev_table, base)?;
+/// println!("DIE at offset: 0x{:x}", parsed_entry.die_offset());
+/// if let Some(parent) = parsed_entry.parent_info() {
+///     println!("Parent entry at: 0x{:x}", parent);
+/// }
+/// ```
+#[derive(Debug, Clone)]
+pub struct ParsedEntry {
+    /// The abbreviation code used for this entry.
+    ///
+    /// References an entry in the abbreviation table that describes
+    /// the structure and attributes of this debug information entry.
+    pub abbrev_code: u64,
+
+    /// The DIE tag for this entry.
+    ///
+    /// Specifies the kind of debugging information entry (e.g., DW_TAG_subprogram,
+    /// DW_TAG_variable, DW_TAG_typedef) as defined in DWARF 5 Section 7.5.
+    pub tag: DwTag,
+
+    /// The offset to the DIE within the compilation unit.
+    ///
+    /// This offset is relative to the start of the compilation unit's
+    /// debug information and can be used to locate the full DIE data.
+    pub die_offset: u32,
+
+    /// Parent information for hierarchical relationships.
+    ///
+    /// Contains the absolute offset of the parent entry in the entry pool
+    /// if this entry has a parent relationship. None indicates either
+    /// no parent (top-level entry) or that parent information is not indexed.
+    pub parent_info: Option<u64>,
```

Comment:
I don't see a list of what attributes are allowed / required / expected for entries in the entry pool.

From 6.1.1.4.8:

> Each index entry in the series begins with an abbreviation code, and is followed
> by the attributes described by the abbreviation declaration for that code. The last
> index entry in the series is followed by a terminating entry whose abbreviation
> code is 0.

So I'm a bit confused by the ones you have added here. Would it be better to expose a more generic way to iterate over the attributes, just like is done for `.debug_info`?